### PR TITLE
Jason/payment details backend

### PR DESCRIPTION
**Pull request by WildlyManifest**
Backend changes done.

### CODE REVIEW

File: client/modules/PaymentDetails/PaymentDetails.js [hunk 1]
```diff
@@ -123,24 +126,37 @@ class PaymentDetails extends Component
...
+          .then(res => console.log(res))
+          .catch(err => console.log(err));
+      }
+      else
+      {
+        // create new creditCard
+        http
+          .client()
+          .post('/paymentDetails/add', {
+            _id: this.state._id,
```
Comment (s3337852) [inline client/modules/PaymentDetails/PaymentDetails.js]: we dont have an _id if we are creating a new credit card

File: client/modules/PaymentDetails/PaymentDetails.js [hunk 2]
```diff
@@ -123,24 +126,37 @@ class PaymentDetails extends Component
...
+            cardNumber: cardNumber,
+            nameOnCard: this.state.nameOnCard,
+            ccv: ccv,
+            expiryMonth: Number(this.state.expiryMonth),
+            expiryYear: Number(this.state.expiryYear),
+          })
+        .then(res => console.log(res))
```
Comment (s3337852) [inline client/modules/PaymentDetails/PaymentDetails.js]: should there be a this.setState({hasCard: true}) here after a successful create? or another call to fetch paymentDetails now that we have an _id.. the controller method returns the user on a successful create but we arent consuming it... also not sure if the creditcard object on the user will be populated without specifically calling populate

File: client/modules/PaymentDetails/PaymentDetails.js (hunk 1 above)
Comment (WildlyManifest) [reply client/modules/PaymentDetails/PaymentDetails.js]: Oops. Corrected.

File: server/controllers/paymentDetails/paymentDetails.controller.js [hunk 3]
```diff
@@ -2,59 +2,85 @@ import User from '../../models/user';
...
+        req.body._id,
+        {
+          cardNumber: req.body.cardNumber,
+          nameOnCard: req.body.nameOnCard,
+          ccv: req.body.ccv,
+          expiryMonth: req.body.expiryMonth,
+          expiryYear: req.body.expiryYear,
+        }
+      )
+    .then((user) => { res.status(200).send(user); })
```
[the remark this answers is not in the document]
Comment (WildlyManifest) [reply server/controllers/paymentDetails/paymentDetails.controller.js]: Done.

File: client/modules/PaymentDetails/PaymentDetails.js (hunk 2 above)
Comment (WildlyManifest) [reply client/modules/PaymentDetails/PaymentDetails.js]: Update card now responds with the card. Both calls will now re-fetch the card Details on success.

**Comment by WildlyManifest** [comment]
Should have done the PR on the feature branch. Anyway, there is a new pull request.

**Pull request closed without merging**